### PR TITLE
build: override queue

### PR DESCRIPTION
**Pull request by Schlagonia**
## Description

Let governance override custom withdraw queues

Fixes # (issue)

## Checklist

- [ ] I have run vyper and solidity linting
- [ ] I have run the tests on my machine
- [ ] I have followed commitlint guidelines
- [ ] I have rebased my changes to the latest version of the main branch

### CODE REVIEW

**Review by poolpitako — APPROVED** [review]

**Comment by Schlagonia** [comment]
> I think this implementation can lead to issues around unexpected outputs.
> 
> if `override_queue` is used and a custom_queue different than default_queue is passed, it should revert. otherwise a user might expect the vault to withdraw from specific strategies and not get that (while that behaviour being important for some reason, e.g. gas usage).
> 
> maybe `override_queue` should be renamed to `allow_custom_queue`
> 
> but I like the idea

I dont think we should revert.

Seems the downside to always reverting is much bigger than servicing a withdraw with a different queue. 

I also dont want to iterate over an custom queue to check if it may be the same as the default.

If someone is concerned enough to be passing a custom queue we should expect them to be able to check the public variable to know if the queue will be overriden.

Question is what is the most obvious nameing/bool to know it will be used. 

I think "allow_custom_queue" could also be misleading since we would allow it to be sent, its just doesnt get used.

**Comment by Schlagonia** [comment]
> > I think this implementation can lead to issues around unexpected outputs.
> > if `override_queue` is used and a custom_queue different than default_queue is passed, it should revert. otherwise a user might expect the vault to withdraw from specific strategies and not get that (while that behaviour being important for some reason, e.g. gas usage).
> > maybe `override_queue` should be renamed to `allow_custom_queue`
> > but I like the idea
> 
> I dont think we should revert.
> 
> Seems the downside to always reverting is much bigger than servicing a withdraw with a different queue.
> 
> I also dont want to iterate over an custom queue to check if it may be the same as the default.
> 
> If someone is concerned enough to be passing a custom queue we should expect them to be able to check the public variable to know if the queue will be overriden.
> 
> Question is what is the most obvious nameing/bool to know it will be used.
> 
> I think "allow_custom_queue" could also be misleading since we would allow it to be sent, its just doesnt get used.

I think it should be 'false' custom queues are used, since thats the default value and the switched needs to be flipped (turned True) to override the queue.

perhaps 'override_custom_queue' would be more clear?

**Comment by Schlagonia** [comment]
> I think this implementation can lead to issues around unexpected outputs.
> 
> if `override_queue` is used and a custom_queue different than default_queue is passed, it should revert. otherwise a user might expect the vault to withdraw from specific strategies and not get that (while that behaviour being important for some reason, e.g. gas usage).
> 
> maybe `override_queue` should be renamed to `allow_custom_queue`
> 
> but I like the idea

I think it should be 'false' custom queues are used, since thats the default value and the switched needs to be flipped (turned True) to override the queue.

perhaps 'override_custom_queue' would be more clear?

File: contracts/VaultV3.vy [hunk 1]
```diff
@@ -700,13 +706,13 @@ def _redeem(
     # funds from strategies.
     if requested_assets > curr_total_idle:
 
-        # Cache the input withdrawal queue.
-        _strategies: DynArray[address, MAX_QUEUE] = strategies
+        # Cache the default queue.
+        _strategies: DynArray[address, MAX_QUEUE] = self.default_queue
```
Comment (fp-crypto) [inline contracts/VaultV3.vy:710]: Copying from storage by default? Good idea?

Comment (Schlagonia) [reply contracts/VaultV3.vy:710]: I think given the expectation that it will almost always use the default_queue its worth it

**Comment by Schlagonia** [comment]
> I think this implementation can lead to issues around unexpected outputs.
> 
> if `override_queue` is used and a custom_queue different than default_queue is passed, it should revert. otherwise a user might expect the vault to withdraw from specific strategies and not get that (while that behaviour being important for some reason, e.g. gas usage).
> 
> maybe `override_queue` should be renamed to `allow_custom_queue`
> 
> but I like the idea

changed the name to `use_default_queue`

**Review by fp-crypto — APPROVED** [review]

**Pull request merged**